src/main/java/uk/gov/digital/ho/hocs/casework/client/auditclient/AuditClient.java

```diff
@@ -233,6 +233,65 @@ public void deleteCaseNoteAudit(CaseNote caseNote) {
         });
     }
 
+    public void viewSomuItemsAudit(UUID caseUUID) {
+        RequestDataDto requestDataDto = RequestDataDto.from(requestData);
+        LocalDateTime localDateTime = LocalDateTime.now();
+        executorService.execute(() -> sendAuditMessage(localDateTime, caseUUID, "", EventType.SOMU_ITEMS_VIEWED, null, requestDataDto.getCorrelationId(),
+                requestDataDto.getUserId(), requestDataDto.getUsername(), requestDataDto.getGroups()));
+    }
+
+    public void viewSomuItemAudit(SomuItem somuItem) {
+        RequestDataDto requestDataDto = RequestDataDto.from(requestData);
+        LocalDateTime localDateTime = LocalDateTime.now();
+        executorService.execute(() -> sendAuditMessage(localDateTime, somuItem.getCaseUuid(), somuItem.getSomuUuid().toString(), EventType.SOMU_ITEM_VIEWED, null, requestDataDto.getCorrelationId(),
+                requestDataDto.getUserId(), requestDataDto.getUsername(), requestDataDto.getGroups()));
+    }
+
+    public void createSomuItemAudit(SomuItem somuItem) {
+        RequestDataDto requestDataDto = RequestDataDto.from(requestData);
+        LocalDateTime localDateTime = LocalDateTime.now();
+        executorService.execute(() -> {
+            String data = "{}";
+            try {
+                data = objectMapper.writeValueAsString(new AuditPayload.SomuItem(somuItem.getUuid(), somuItem.getCaseUuid(), somuItem.getSomuUuid(), somuItem.getData()));
+            } catch (JsonProcessingException e) {
+                logFailedToParseDataPayload(e);
+            }
+            sendAuditMessage(localDateTime, somuItem.getCaseUuid(), data, EventType.SOMU_ITEM_CREATED, null,
```

Comment:
We need to change the calls to use the sendAuditMessage that includes the data in the audit data, that will allow for search to add it to indexing on Elastic Search. 

This needs changing here and in the update and delete commands.

Reply:
This is now resolved.